src/middleware.js

```diff
@@ -1,12 +1,32 @@
 import * as defaultOptions from './defaults';
 import { getActionTypes } from './getActionTypes';
+import axios from 'axios';
```

Comment:
We don't need to import axios, we can bind interceptors to client.
I suggest:
```
function bindInterceptors(client, getState, interceptors) {
  //...
  client.interceptors.request.use(interceptor.bind(null, getState));
  //...
}
```
This way we can bind it to single axios instance, because we can have more of them.